src/types.rs

```diff
@@ -153,6 +156,7 @@ impl Display for Type {
             Type::MetadataType => write!(f, "metadata"),
             Type::LabelType => write!(f, "label"),
             Type::TokenType => write!(f, "token"),
+            Type::TargetExtType => write!(f, "target"),
```

Comment:
This isn't quite right, since target types can take subtypes and integer values.